packages/components/src/unit-control/index.tsx

```diff
@@ -260,23 +196,20 @@ function UnforwardedUnitControl(
 		<Root className="components-unit-control-wrapper" style={ style }>
 			<ValueInput
 				aria-label={ label }
-				type={ isPressEnterToChange ? 'text' : 'number' }
 				{ ...omit( props, [ 'children' ] ) }
 				autoComplete={ autoComplete }
 				className={ classes }
 				disabled={ disabled }
 				disableUnits={ disableUnits }
 				isPressEnterToChange={ isPressEnterToChange }
 				label={ label }
-				onBlur={ handleOnBlur }
```

Comment:
I've made sure to keep the behavior from #39589 (as the tests added that still pass confirm). I didn't add the prop back here as it can pass through as part of `props`. The unit select takes it up like so:
https://github.com/WordPress/gutenberg/blob/9a0a708bfbb14ec78ffa9204c7eaffc3a2d21fc7/packages/components/src/unit-control/index.tsx#L181

If you'd prefer that `onBlur` is destructured from `unitControlProps` it'd be fine by me.